platform/darwin/src/NSExpression+MGLAdditions.h

```diff
@@ -42,6 +90,133 @@ NS_ASSUME_NONNULL_BEGIN
+
+/**
+ Returns a step function expression specifying the function operator, default color
+ and stops.
+ 
+ @param function The operator type in which this expression is applied.
+ @param color The default color.
+ @param stops The stops dictionay must be numeric literals in strictly ascending order.
+ */
++ (instancetype)mgl_expressionForStepFunction:(nonnull MGLExpressionStyleFunction)function defaultColor:(nonnull MGLColor *)color stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops;
+
+/**
+ Returns a step function expression specifying the function operator, default expression
+ and stops.
+ 
+ @param operatorExpression The operator expression.
+ @param expression The expression which could be a constant or function expression.
+ @param stops The stops dictionay must be numeric literals in strictly ascending order.
+ */
++ (instancetype)mgl_expressionForStepFunction:(nonnull NSExpression*)operatorExpression defaultExpression:(nonnull NSExpression *)expression stops:(nonnull NSExpression*)stops;
+
+/**
+ Returns an interpolated function expression specifying the function operator, curve type
+ and steps.
+ 
+ @param function The operator type in which this expression is applied.
+ @param curveType The curve type could be `MGLExpressionInterpolationModeLinear`,
+                    `MGLExpressionInterpolationModeExponential` and
+                    `MGLExpressionInterpolationModeCubicBezier`.
+ @param steps The steps dictionay must be numeric literals in strictly ascending order.
+ */
++ (instancetype)mgl_expressionForInterpolateFunction:(nonnull MGLExpressionStyleFunction)function curveType:(nonnull MGLExpressionInterpolationMode)curveType steps:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)steps;
+
+/**
+ Returns an interpolated function expression specifying the function operator, curve type,
+ parameters and steps.
+ 
+ @param expressionOperator The expression operator.
+ @param curveType The curve type could be `MGLExpressionInterpolationModeLinear`,
+ `MGLExpressionInterpolationModeExponential` and
+ `MGLExpressionInterpolationModeCubicBezier`.
+ @param parameters The parameters expression.
+ @param steps The steps expression.
+ */
++ (instancetype)mgl_expressionForInterpolateFunction:(nonnull NSExpression*)expressionOperator curveType:(nonnull MGLExpressionInterpolationMode)curveType parameters:(nullable NSExpression *)parameters steps:(nonnull NSExpression*)steps;
+
+/**
+ Returns a string constant expression appending the passed string.
+ 
+ @param string The string to append.
+ */
+- (instancetype)mgl_appendingString:(NSString *)string;
+
+/**
+ Returns a string constant expression appending the passed expression.
+ 
+ @param expression The evaluated expression must return a string.
+ */
+- (instancetype)mgl_appendingExpression:(NSExpression *)expression;
```

Comment:
Per Objective-C naming conventions, this method should be named `-mgl_expressionByAppendingExpression:` (cf. `-[NSString stringByAppendingString:]`). Specify a Swift name of `mgl_appending(_:)`.